File: Data Structures & Algorithms/meeting-schedule/submission-2.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def canAttendMeetings(self, intervals: List[Interval]) -> bool:
        n = len(intervals)
        if n <= 1:
            return True
        intervals.sort(key = lambda x : x.start)
        maxEnd = intervals[0].end
        for i in range(1, n):
            if intervals[i].start < maxEnd:
                return False
            else:
                maxEnd = max(maxEnd, intervals[i].end)
        return True
        

File: Data Structures & Algorithms/is-anagram/submission-2.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int n = s.length();
        int m = t.length();
        if(n != m){
            return false;
        }
        vector<char> sc(s.begin(), s.end());
        vector<char> tc(t.begin(), t.end());
        sort(sc.begin(), sc.end());
        sort(tc.begin(), tc.end());
        for(int i = 0; i < n; i++){
            if (sc[i] != tc[i]){
                return false;
            }
        }
        return true;

    }
};
